File: ruby/lib/casilla_juez.rb
module Civitas
    class CasillaJuez < Casilla
        def initialize(carcel)
            super("Juez")
            @carcel=carcel
        end
        attr_reader :carcel

        def recibeJugador(actual, todos)
            if jugadorCorrecto(actual, todos)
                informe(actual, todos)
                todos[actual].encarcelar(@carcel)
            end
        end
    end
end

File: Civitas_gui/src/civitas/CasillaJuez.java
package civitas;
import java.util.ArrayList;

public class CasillaJuez extends Casilla {
    private static int carcel;
    CasillaJuez(int numCasillaCarcel, String nombre) {
        super(nombre);
        carcel = numCasillaCarcel;
    }
    
    @Override
    public String getTipo(){
        return "Juez";
    }

    @Override
    public void recibeJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            todos.get(actual).encarcelar(carcel);
        }
    }
}